clang/lib/Format/FormatToken.cpp

```diff
@@ -34,9 +34,15 @@ const char *getTokenTypeName(TokenType Type) {
   return nullptr;
 }
 
+// Sorted common C++ non-keyword types.
+static SmallVector<StringRef> CppNonKeywordTypes = {
+    "byte",   "int16_t",  "int32_t",  "int64_t",  "int8_t",
+    "size_t", "uint16_t", "uint32_t", "uint64_t", "uint8_t",
```

Comment:
Maybe also `ptrdiff_t`, `intptr_t`, `uintptr_t`. And there's also `float_t`, `double_t`, `uint_least32_t`, ...

I wonder if the best heuristic to use here would be simply to look for an identifier that matches `[a-z0-9_]*_t`.

Reply:
As the comment says, the list includes ~common~ non-keyword types. We can always grow the list in the future if necessary. I thought about using regex but went for a specific list in order to minimize the possibility of regression.

Reply:
According to codesearch.isocpp.org, `intptr_t`, `uintptr_t`, and `ptrdiff_t` are all more common than `int8_t`. Also I'd note that POSIX reserves all identifiers ending in `_t` for use as types, and C reserves all identifiers starting with `int` or `uint` and ending in `_t` for use as types.

Reply:
I was aware of that. By “common” I mean common C-casted. We definitely don’t want to include POSIX-specific types as we have the TypeNames option for that. Also, we don’t want to cater to C-specific constructs if it might regress C++ code.

Reply:
I considered to include all [fixed width integer types](https://en.cppreference.com/w/cpp/types/integer) but decided to include only the common ones to avoid possible regressions. (I also looked at [fixed width floating-point types](https://en.cppreference.com/w/cpp/types/floating-point) and think we should wait until C++23 is commonly used.)

Reply:
On codesearch.isocpp.org, `(intptr_t)` is more common than `(int8_t)` and `(int16_t)`, and similar to `(int32_t)`. But why should this general function that's used in various places in clang-format care about how often the type is cast with a C-style cast? That doesn't make sense to me.

I find it extremely unlikely that anyone would use `intN_t` as anything other than a type; I think your concerns about regressions are unfounded. In contrast, there are 518 hits for `(intptr_t)&` in isocpp codesearch, which seems pretty logical to me because casts to `(intptr_t)` will be casts from pointers.

Reply:
> I find it extremely unlikely that anyone would use `intN_t` as anything other than a type; I think your concerns about regressions are unfounded.

"Extremely unlikely" doesn't mean impossible.

I actually hesitated on whether to include `intptr_t`/`uintprt_t` because they are marked as optional. I can include them though.